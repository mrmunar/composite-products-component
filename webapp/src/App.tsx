import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';
import { Provider } from "react-redux";

import Container from './components/containers/main';
import AddEditCompositeProducts from './pages/CompositeProducts/AddEdit';
import ListCompositeProducts from './pages/CompositeProducts/List';
import store from './redux/store';

import './App.css';

function App() {
  return (
    <div className="App">
      <Provider store={store}>
        <Container>
          <Router>
            <Switch>
              <Route path="/add" component={AddEditCompositeProducts} />
              <Route path="/:id" component={AddEditCompositeProducts} />
              <Route path="/" component={ListCompositeProducts} />
            </Switch>
          </Router>
        </Container>
      </Provider>
    </div>
  );
}

export default App;
