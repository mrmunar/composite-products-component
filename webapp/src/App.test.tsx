import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('should take a snapshot', () => {
  const { baseElement } = render(<App />);
  expect(baseElement).toMatchSnapshot();
});
