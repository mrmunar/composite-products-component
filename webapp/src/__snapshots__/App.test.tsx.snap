// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="main-container"
      >
        <div
          class="row justify-content-center"
        >
          <div
            class="col-lg-6"
          >
            <div
              class="text-left"
            >
              <div
                class="text-left"
              >
                <h3
                  data-testid="title"
                >
                  List Composite Products
                </h3>
                <div
                  class="mb-3"
                >
                  <button
                    class="btn btn-info"
                    data-testid="button"
                  >
                    Create
                  </button>
                </div>
                Loading...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
